airflow-core/src/airflow/api_fastapi/app.py

```diff
@@ -96,6 +96,7 @@ def create_app(apps: str = "all") -> FastAPI:
         init_views(app)  # Core views need to be the last routes added - it has a catch all route
         init_error_handlers(app)
         init_middlewares(app)
+        init_generic_middlewares(app)
```

Comment:
I feel it confusing to have `init_middlewares` and `init_generic_middlewares`. I think it would be better to add the new middleware in the existing `init_middlewares`